pkg/containerinstall/install.go

```diff
@@ -0,0 +1,255 @@
+
+	_, err := runContainer(m.conn, m.log, s)
+	return err
+}
+
+func (m *manager) containerFinished(context.Context) (bool, error) {
+	containerName := "installer-" + m.doc.ID
+	inspectData, err := containers.Inspect(m.conn, containerName, nil)
+	if err != nil {
+		return false, err
+	}
+
+	if inspectData.State.Status == "exited" || inspectData.State.Status == "stopped" {
+		if inspectData.State.ExitCode != 0 {
+			getContainerLogs(m.conn, m.log, containerName)
+			return true, fmt.Errorf("container exited with %d", inspectData.State.ExitCode)
+		} else {
+			m.success = true
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (m *manager) writeFiles(ctx context.Context) error {
+	encCluster, err := json.Marshal(m.doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+	_, err = secrets.Create(m.conn, bytes.NewBuffer(encCluster), &secrets.CreateOptions{Name: to.StringPtr(m.doc.ID + "-99_aro.json")})
+	if err != nil {
+		return err
+	}
+
+	encSub, err := json.Marshal(m.sub.Subscription)
+	if err != nil {
+		return err
+	}
+	_, err = secrets.Create(m.conn, bytes.NewBuffer(encSub), &secrets.CreateOptions{Name: to.StringPtr(m.doc.ID + "-99_sub.json")})
+	if err != nil {
+		return err
+	}
+
+	if m.isDevelopment {
+		basepath := os.Getenv("ARO_CHECKOUT_PATH")
+		if basepath == "" {
+			// This assumes we are running from an ARO-RP checkout in development
+			var err error
+			_, curmod, _, _ := runtime.Caller(0)
+			basepath, err = filepath.Abs(filepath.Join(filepath.Dir(curmod), "../.."))
+			if err != nil {
+				return err
+			}
+		}
+
+		err = m.secretFile(filepath.Join(basepath, "secrets/proxy.crt"), "proxy.crt")
+		if err != nil {
+			return err
+		}
+
+		err = m.secretFile(filepath.Join(basepath, "secrets/proxy-client.crt"), "proxy-client.crt")
+		if err != nil {
+			return err
+		}
+
+		err = m.secretFile(filepath.Join(basepath, "secrets/proxy-client.key"), "proxy-client.key")
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *manager) secretFile(from, name string) error {
+	f, err := os.Open(from)
+	if err != nil {
+		return err
+	}
+
+	_, err = secrets.Create(m.conn, f, &secrets.CreateOptions{Name: to.StringPtr(m.doc.ID + "-" + name)})
+	return err
+}
+
+func (m *manager) cleanupContainers(ctx context.Context) error {
+	containerName := "installer-" + m.doc.ID
+
+	if !m.success {
+		m.log.Infof("cleaning up failed container %s", containerName)
+		getContainerLogs(m.conn, m.log, containerName)
+	}
+
+	_, err := containers.Remove(m.conn, containerName, &containers.RemoveOptions{Force: to.BoolPtr(true), Ignore: to.BoolPtr(true)})
+	if err != nil {
+		m.log.Errorf("unable to remove container: %v", err)
+	}
+
+	for _, secretName := range []string{"99_aro.json", "99_sub.json", "proxy.crt", "proxy-client.crt", "proxy-client.key"} {
+		err = secrets.Remove(m.conn, m.doc.ID+"-"+secretName)
+		if err != nil {
```

Comment:
would it make sense to not log anything if the error is `isNotFound` or something?

Reply:
I set it to debug for now